src/proteomes/components/entry/ComponentsButtons.tsx

```diff
@@ -0,0 +1,112 @@
+import { FC, useState, Suspense, useCallback } from 'react';
+import { Button, DownloadIcon } from 'franklin-sites';
+import { Link } from 'react-router-dom';
+
+import SlidingPanel, {
+  Position,
+} from '../../../shared/components/layouts/SlidingPanel';
+
+import useLocalStorage from '../../../shared/hooks/useLocalStorage';
+
+import lazy from '../../../shared/utils/lazy';
+
+import localStorageKeys from '../../../app/config/localStorageKeys';
+import { Column, nsToDefaultColumns } from '../../../shared/config/columns';
+import { LocationToPath, Location } from '../../../app/config/urls';
+
+import { Namespace } from '../../../shared/types/namespaces';
+import { ProteomesAPIModel } from '../../adapters/proteomesConverter';
+
+const DownloadComponent = lazy(
+  () =>
+    import(
+      /* webpackChunkName: "download" */ '../../../shared/components/download/Download'
+    )
+);
+
+const ComponentsButtons: FC<
+  Pick<ProteomesAPIModel, 'id'> & {
+    nComponents: number;
+    selectedEntries: string[];
+  }
+> = ({ id, nComponents, selectedEntries }) => {
+  const [displayDownloadPanel, setDisplayDownloadPanel] = useState(false);
+  const query = '';
+  const defaultColumns = nsToDefaultColumns[Namespace.uniprotkb];
+  const [tableColumns] = useLocalStorage<Column[]>(
+    localStorageKeys.tableColumns(Namespace.uniprotkb),
+    defaultColumns || []
+  );
+
+  const getViewButtonQuery = useCallback(
+    () =>
+      `query=(proteome:${id})${
+        selectedEntries.length !== 0 && selectedEntries.length !== nComponents
+          ? ` AND (${selectedEntries
+              .map((component) => `proteomecomponent:"${component}"`)
+              .join(' OR ')})`
+          : ''
+      }`,
+    [nComponents, id, selectedEntries]
+  );
+
+  const getViewButtonText = useCallback(() => {
+    const nSelected = selectedEntries.length;
+    if (nComponents === 1) {
+      return 'View UniProtKB entry for component';
+    }
+    return `View UniProtKB ${nSelected !== 1 ? 'entries' : 'entry'} for ${
+      nSelected === 0 || nSelected === nComponents
+        ? `all ${nComponents}`
+        : nSelected || ''
+    } ${nSelected && nSelected !== nComponents ? 'selected' : ' '} ${
+      nSelected === 0 || nSelected > 1 ? 'components' : 'component'
+    }`;
+  }, [nComponents, selectedEntries.length]);
+
+  return (
+    <>
+      {displayDownloadPanel && (
+        <Suspense fallback={null}>
+          <SlidingPanel
+            position={Position.left}
+            onClose={() => setDisplayDownloadPanel(false)}
+            yScrollable
+          >
+            <DownloadComponent
```

Comment:
The "Preview n" functionality doesn't seem to work. For FASTA nothing happens and for table outputs I just get the table headers.

Reply:
Fixed this. I am relying on protein count which isn't available yet so have hard coded some values as a fallback for now (with TODO to eventually remove). I had to adapt the Download component a bit to accommodate downloading the selected proteome components within the uniprotkb namespace.